pkg/flist/cleanup.go

```diff
@@ -15,13 +16,44 @@ import (
 // Cleaner interface, implementer of this interface
 // can start a cleaner job
 type Cleaner interface {
-	// MountsCleaner runs the clean process, MountsCleaner should be
-	// blocking. Caller then can do `go MountsCleaner()` to run it in the background
+	MountsCleaner(ctx context.Context, every time.Duration)
+	// CacheCleaner runs the clean process, CacheCleaner should be
+	// blocking. Caller then can do `go CacheCleaner()` to run it in the background
 	CacheCleaner(ctx context.Context, every time.Duration, age time.Duration)
 }
 
 var _ Cleaner = (*flistModule)(nil)
 
+func (f *flistModule) MountsCleaner(ctx context.Context, every time.Duration) {
```

Comment:
This is very unsafe procedures. 

this runs in the background in a continues loop. What prevents this from deleting a mountpoint that was just created, just before the filesystem is actually mounted. 

Hence this procedure can cause race conditions and errors that are impossible to debug.